### PR TITLE
fix nvidia-oot package

### PR DESCRIPTION
**Pull request by Mic92**
Test with:
$ nix-build ./packages.nix -A nvidia-oot-cross

### CODE REVIEW

File: nvidia-oot.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    export KERNEL_HEADERS=$(pwd)/linux-dev/lib/modules/${linux.modDirVersion}/build
+
+  '';
+
+  nativeBuildInputs = linux.moduleBuildDependencies ++ [ ];
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+
+  buildPhase = ''
+    export CROSS_COMPILE=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}
```
Comment (Mic92) [inline nvidia-oot.nix:61]:
```suggestion
```

File: nvidia-oot.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    make $makeFlags
```
Comment (Mic92) [inline nvidia-oot.nix:62]: Why might not need this build phase at all actually...

File: nvidia-oot.nix [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  '';
```
Comment (Mic92) [inline nvidia-oot.nix:63]:
```suggestion
```

File: nvidia-oot.nix [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  # cd source
+  # git init .
+  # git add .
+  # git commit -m "Initial commit"
+  # <make changes>
+  # git diff > ../0001-build-fixes.patch
+  patches = [ ./0001-build-fixes.patch ];
+
+  postUnpack = ''
+    cp -r ${linux.dev} linux-dev
```
Comment (Mic92) [inline nvidia-oot.nix:50]:
```suggestion
    # make kernel headers readable for the nvidia build system.
    cp -r ${linux.dev} linux-dev
```

File: nvidia-oot.nix [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  hardeningDisable = [ "pic" ];
+
+  # unclear why we need to add nvidia-oot/sound/soc/tegra-virt-alt/include
+  # this only happens in the nix-sandbox and not in the nix-shell
+  NIX_CFLAGS_COMPILE = "-fno-stack-protector -Wno-error=attribute-warning -I ${source}/nvidia-oot/sound/soc/tegra-virt-alt/include ${
+    lib.concatMapStrings (x: "-isystem ${x} ") (kernelIncludes linux.dev)
+  }";
+
+  installTargets = [ "modules_install" ];
+  enableParallelBuilding = false;
```
Comment (Mic92) [inline nvidia-oot.nix:88]: I think this is not doing actually anything. I just had a hunch that I encountered a race condition:

```suggestion
```

File: 0001-build-fixes.patch [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
++		NV_KERNEL_SOURCES=$(KERNEL_HEADERS)/source \
+ 		NV_KERNEL_OUTPUT=$(KERNEL_OUTPUT) \
+ 		-f $(NVIDIA_CONFTEST)/nvidia/Makefile
+ endif
+@@ -110,7 +110,7 @@ define display-cmd
+ 		LOCALVERSION=$(version) \
+ 		NV_VERBOSE=0 \
+ 		KERNELRELEASE="" \
+-		SYSSRC=$(NVIDIA_HEADERS) \
++		SYSSRC=$(NVIDIA_HEADERS)/source \
```
Comment (Mic92) [inline 0001-build-fixes.patch:19]: This one could potentially be dropped?

File: nvidia-oot.nix [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  postUnpack = ''
+    cp -r ${linux.dev} linux-dev
+    chmod -R u+w linux-dev
+    export KERNEL_HEADERS=$(pwd)/linux-dev/lib/modules/${linux.modDirVersion}/build
+
+  '';
+
+  nativeBuildInputs = linux.moduleBuildDependencies ++ [ ];
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
```
Comment (Mic92) [inline nvidia-oot.nix:58]:
```suggestion
  # some calls still go to `gcc` in the build
  depsBuildBuild = [ buildPackages.stdenv.cc ];
```

**Comment by TanelDettenborn** [comment]
@Mic92 Thank you!

Are you going to push more changes? If busy then I could merge this pull request and implement your suggestion in another commit?

**Comment by Mic92** [comment]
@TanelDettenborn I might do not get to that today. So please go ahead with the changes.

**Pull request merged**